dsc_lib/src/dscresources/dscresource.rs

```diff
@@ -363,28 +373,176 @@ pub fn get_diff(expected: &Value, actual: &Value) -> Vec<String> {
             if value.is_object() {
                 let sub_diff = get_diff(value, &actual[key]);
                 if !sub_diff.is_empty() {
+                    debug!("diff: sub diff for {key}");
                     diff_properties.push(key.to_string());
                 }
             }
             else {
-                match actual.as_object() {
-                    Some(actual_object) => {
-                        if actual_object.contains_key(key) {
-                            if value != &actual[key] {
+                // skip `$schema` key as that is provided as input, but not output typically
+                if key == "$schema" {
+                    continue;
+                }
+
+                if let Some(actual_object) = actual.as_object() {
+                    if actual_object.contains_key(key) {
+                        if let Some(value_array) = value.as_array() {
+                            if let Some(actual_array) = actual[key].as_array() {
+                                if !is_same_array(value_array, actual_array) {
+                                    info!("diff: arrays differ for {key}");
+                                    diff_properties.push(key.to_string());
+                                }
+                            } else {
+                                info!("diff: {} is not an array", actual[key]);
                                 diff_properties.push(key.to_string());
                             }
-                        }
-                        else {
+                        } else if value != &actual[key] {
                             diff_properties.push(key.to_string());
                         }
-                    },
-                    None => {
+                    } else {
+                        info!("diff: {key} missing");
                         diff_properties.push(key.to_string());
-                    },
+                    }
+                } else {
+                    info!("diff: {key} not object");
+                    diff_properties.push(key.to_string());
                 }
             }
         }
     }
 
     diff_properties
 }
+
+/// Compares two arrays independent of order
+fn is_same_array(expected: &Vec<Value>, actual: &Vec<Value>) -> bool {
+    if expected.len() != actual.len() {
+        info!("diff: arrays are different lengths");
+        return false;
+    }
+
+    for item in expected {
+        if !array_contains(actual, item) {
+            info!("diff: actual array missing expected element");
+            return false;
+        }
+    }
+
+    true
+}
+
+fn array_contains(array: &Vec<Value>, find: &Value) -> bool {
+    for item in array {
+        if find.is_boolean() && item.is_boolean() && find.as_bool().unwrap() == item.as_bool().unwrap() {
```

Comment:
would it be more efficient to nest the checks as an if/else? 
a worst case example - `find=false` and `item=true`

Reply:
Since the comparison is left to right, if any fail, then no additional processing is needed.  For example, if `find` is NOT a boolean, the latter checks aren't executed.  If we have a nested `if` statement, then it'll be slightly more expensive as it starts a new execution of an `if` statement rather than just completing the previous one.

I actually started with nested `if` and clippy complained :)